packages/dds/sequence/src/sharedStringWithSequentialId.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IMergeTreeOp, Marker,
+    MergeTreeDeltaType, reservedMarkerIdKey } from "@fluidframework/merge-tree";
+import { ISequencedDocumentMessage, MessageType } from "@fluidframework/protocol-definitions";
+import { parseHandles } from "@fluidframework/shared-object-base";
+import { createIdAfterMin, createIdBeforeMax } from "./generateSequentialId";
+import { SharedString } from "./sharedString";
+
+export function sharedStringWithSequentialIdMixin(Base: typeof SharedString = SharedString): typeof SharedString {
+    return class SequenceWithSequentialId extends Base {
+        protected processMergeTreeMsg(
+            rawMessage: ISequencedDocumentMessage): void {
+                const message = parseHandles(
+                    rawMessage,
+                    this.runtime.IFluidSerializer);
+                this.applySequentialId(message);
```

Comment:
Q to @anthony-murphy: That's probably alright, but if there was a way to not depend on op format, but listen on delta events, and modify internal state of merge tree without causing ops to flow, that feels like it would be a bit more sustainable approach (at least in terms of contracts).

Reply:
at a minimum we should not overwrite processMergeTreeMsg, we should provide a new function specifically for mixin that is called by processMergeTreeMsg. But i think what you suggest is correct. I have ideas here i can share when we meet

---
In reply to: [545994771](https://github.com/microsoft/FluidFramework/pull/4641#discussion_r545994771) [](ancestors = 545994771)

Reply:
Resolving as this is oudated... we are no longer using processMergeTreeMsg